src/lightning_lite/utilities/data.py

```diff
@@ -360,6 +360,12 @@ def _replace_dunder_methods(base_cls: Type, store_explicit_arg: Optional[str] =
 
     It patches the ``__init__``, ``__setattr__`` and ``__delattr__`` methods.
     """
+    _patch_dunder_methods(base_cls, store_explicit_arg)
+    yield
+    _unpatch_dunder_methods(base_cls)
```

Comment:
IMO we should remove this and go all-in into the import-level patch by having PL use it